From 032c03e8dd146b6b9d4d535ad533f162d5f7dc6e Mon Sep 17 00:00:00 2001
From: "Jaikrishna, Nemallapudi" <nemallapudi.jaikrishna@intel.com>
Date: Sun, 1 Oct 2023 16:35:08 +0530
Subject: [PATCH] Fix sepolicy issues of camera HAL

Need access to /sys/class/video4linux/ for reading the device name
 -Added sysfs video device access policy

Need access to netlink socket to communicate with loopback driver
 -Added netlink socket related policy

Tracked-On: OAM-112466
Signed-off-by: Jaikrishna, Nemallapudi <nemallapudi.jaikrishna@intel.com>
Signed-off-by: shivasku82 <shiva.kumara.rudrappa@intel.com>
---
 camera-ext/ivi/hal_camera_default.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/camera-ext/ivi/hal_camera_default.te b/camera-ext/ivi/hal_camera_default.te
index aca2051..8e90705 100644
--- a/camera-ext/ivi/hal_camera_default.te
+++ b/camera-ext/ivi/hal_camera_default.te
@@ -7,4 +7,7 @@ allow hal_camera_default hal_graphics_allocator_default_tmpfs:file { map read wr
 allow hal_camera_default hal_graphics_mapper_hwservice:hwservice_manager find;
 allow hal_camera_default hal_graphics_composer_default:fd use;
 allow hal_camera_default self:{ socket vsock_socket } { create read write listen accept bind };
+allow hal_camera_default self:netlink_socket { create bind setopt };
+allow hal_camera_default sysfs_video:dir r_dir_perms;
+allow hal_camera_default sysfs_video:file rw_file_perms;
 set_prop(hal_camera_default, vendor_camera_default_prop)
-- 
2.40.0

